### DISCUSSION TITLE
Are Codespaces Helpful for Native GUI Application Development?

### DISCUSSION THREAD

**Discussion by JackEAllen** [community/events#42] [Cloud]
I can definitely see the benefits of Codespaces for many development situations, but I often find myself wondering if there is any benefit currently or in the future for developers who work with native GUI applications where developers need to spin up instances of the native application as part of their development process.

In the scientific software development industry, I regularly come across and also work with native applications using front end technologies / frameworks such as PyQT or a Java Eclipse RCP where often, it makes more sense for a native application over a web app to be developed.

Is there anyone here who uses Codespaces for developing native GUI applications or can see any potential benefit to doing so?

**Comment by ossanna16** [comment]
Great question @JackEAllen!

**Comment by dewinterjack** [comment]
I have been working on an electron app recently and have not seen other devs using codespaces with it yet.

You could add [desktop-lite](https://github.com/devcontainers/features/tree/main/src/desktop-lite) as a feature to your devcontainer and access the desktop GUI through VNC Viewer or the browser. I'm not sure how smooth it would feel but I would be interested to give this a go.

**Reply by JackEAllen** [reply]
Definitely a possibility using something like desktop-lite if it can be consistently stable. I usually end up remoting into a given machine I have in the office using something like [NoMachine](https://www.nomachine.com/) to see the GUI and VS Code using the [Remote Development](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) extension from my local machine at the same time to work on the code
